src/HotChocolate/Data/test/Data.Filters.Tests/Expression/QueryableFilterVisitorMethodTests.cs

```diff
@@ -65,81 +81,86 @@ public void Create_MethodComplex_Expression()
             Assert.False(funcFalse(b));
         }
 
-        public class QueryableSimpleMethodTest
+        private class QueryableSimpleMethodTest
             : QueryableDefaultFieldHandler
         {
-            public static MethodInfo Method = typeof(Foo).GetMethod(nameof(Foo.Simple))!;
+            private static readonly MethodInfo Method = typeof(Foo).GetMethod(nameof(Foo.Simple))!;
+
+            private IExtendedType? _extendedType;
 
             public override bool CanHandle(
                 ITypeDiscoveryContext context,
                 FilterInputTypeDefinition typeDefinition,
-                FilterFieldDefinition fieldDefinition) =>
-                fieldDefinition is FilterOperationFieldDefinition {Id: 156 };
+                FilterFieldDefinition fieldDefinition)
+            {
+                _extendedType ??= context.TypeInspector.GetReturnType(Method);
```

Comment:
@michaelstaib  Maybe you have a better idea. This currently looks like a hack. Should we bring back the initialization of the Handlers?
We need a type reference of this method to push onto the stack. We used the member before. but now we dont have a member

Reply:
Nope ... since we have dependency injection magic :) you can inject the `IConventionContext` into the constructor. For convenience, I have reworked it so that you can inject `ITypeInspector` directly.

```
protected override void OnComplete(
            IConventionContext context,
            FilterProviderDefinition definition)
        {
            if (definition.Handlers.Count == 0)
            {
                throw FilterProvider_NoHandlersConfigured(this);
            }

            IServiceProvider services = new DictionaryServiceProvider(
                (typeof(IFilterProvider), this),
                (typeof(IConventionContext), context),
                (typeof(IDescriptorContext), context.DescriptorContext),
                (typeof(ITypeInspector), context.DescriptorContext.TypeInspector))
                .Include(context.Services);

            foreach ((Type Type, IFilterFieldHandler? Instance) handler in definition.Handlers)
            {
                switch (handler.Instance)
                {
                    case null when services.TryGetOrCreateService(
                        handler.Type,
                        out IFilterFieldHandler<TContext>? service):
                        _fieldHandlers .Add(service);
                        break;
                    case null:
                        context.ReportError(
                            FilterProvider_UnableToCreateFieldHandler(this, handler.Type));
                        break;
                    case IFilterFieldHandler<TContext> casted:
                        _fieldHandlers .Add(casted);
                        break;
                }
            }
        }
```

So, you basically have all the schema level services plus these 4 context bound services.